### PR TITLE
Bugfix/templates

### PR DESCRIPTION
**Pull request by fortmarek**
Resolves https://github.com/tuist/tuist/issues/1485

### Short description 📝

If you wanted to describe multiple templates, you would get a compiler error when trying to build a project generated by `tuist edit` since every template mandated `Template.swift` which would then result in multiple files with the same name.

### Solution 📦

Going forward, `Template.swift` should be named `name_of_template.swift` to avoid any collision. We don't want this to be a breaking change, so `Template.swift` will be allowed for the time being.
I had to introduce a `path` parameter for `fileName` function in `Manifest` to extract the `baseNameWithoutExt` for `template` case. Ideally, this could be done by adding associated value to `template`, but that would break `CaseIterable` conformance, so I figured this might be better in the end (and in the future if a `Manifest` needs to leverage `path`, it's gonna be there)

### Implementation 👩‍💻👨‍💻

- [x] Change default template naming
- [x] Modify fixture to have two templates
- [x] Fix/add tests
- [x] Update documentation
- [x] Update Changelog

### CODE REVIEW

**Comment by fortmarek** [comment]
There's a new rule in `swiftformat` enabled by default: https://github.com/nicklockwood/SwiftFormat/pull/668 which makes this build fail

Do we want it or not? @tuist/core ?

File: features/step_definitions/scaffold.rb [hunk 1]
```diff
@@ -7,3 +7,7 @@
 Then(/content of a file named (.+) in a directory (.+) should be equal to (.+)/) do |file, dir, content|
     assert_equal File.read(File.join(@dir, dir, file)), content
 end
+
+Then(/content of a file named (.+) in a directory (.+) should be equal to:$/) do |file, dir, content|
```
Comment (pepibumur) [inline features/step_definitions/scaffold.rb:11]: mm, I might be wrong, but I don't see a capturing group for `content` in the sentence. Should it be?

```rb
/content of a file named (.+) in a directory (.+) should be equal to: (.+)$/
```

Comment (fortmarek) [reply features/step_definitions/scaffold.rb:11]: The syntax seems to be different for multiline strings: http://asymmetrical-view.com/2011/06/02/cucumber-gherkin-and-multiline-arguments.html

**Comment by pepibumur** [comment]
> There's a new rule in swiftformat enabled by default: nicklockwood/SwiftFormat#668 which makes this build fail

Let's enable it

**Pull request merged**